pkgs/os-specific/darwin/pam-reattach/default.nix

```diff
@@ -19,10 +19,10 @@ stdenv.mkDerivation rec {
         "arm64"
     }"
     "-DENABLE_CLI=ON"
-  ] ++ lib.optional (!stdenv.isAarch64) "-DCMAKE_LIBRARY_PATH=${darwin.apple_sdk.sdk}/usr/lib";
+  ] ++ lib.optional  (darwin.apple_sdk ? sdk) "-DCMAKE_LIBRARY_PATH=${darwin.apple_sdk.sdk}/usr/lib";
 
   buildInputs = [ openpam ]
-    ++ lib.optional (!stdenv.isAarch64) darwin.apple_sdk.sdk;
+    ++ lib.optional (darwin.apple_sdk ? sdk) darwin.apple_sdk.sdk;
```

Comment:
pam-reattach builds for me without adding the SDK, so I think it can be dropped. `CMAKE_OSX_ARCHITECTURES` can be defined as `"-DCMAKE_OSX_ARCHITECTURES=${stdenv.hostPlatform.darwinArch}"` (but GitHub won’t let me comment on those lines). This is the patch I used locally to build it (with some extra changes to `cmakeFlags`):

```patch
diff --git a/pkgs/os-specific/darwin/pam-reattach/default.nix b/pkgs/os-specific/darwin/pam-reattach/default.nix
index 4350865080f0..4a20d7dd93f8 100644
--- a/pkgs/os-specific/darwin/pam-reattach/default.nix
+++ b/pkgs/os-specific/darwin/pam-reattach/default.nix
@@ -12,17 +12,11 @@ stdenv.mkDerivation rec {
   };
 
   cmakeFlags = [
-    "-DCMAKE_OSX_ARCHITECTURES=${
-      if stdenv.hostPlatform.system == "x86_64-darwin" then
-        "x86_64"
-      else
-        "arm64"
-    }"
-    "-DENABLE_CLI=ON"
-  ] ++ lib.optional (!stdenv.isAarch64) "-DCMAKE_LIBRARY_PATH=${darwin.apple_sdk.sdk}/usr/lib";
+    (lib.cmakeFeature "CMAKE_OSX_ARCHITECTURES" stdenv.hostPlatform.darwinArch)
+    (lib.cmakeBool "ENABLE_CLI" true)
+  ];
 
-  buildInputs = [ openpam ]
-    ++ lib.optional (!stdenv.isAarch64) darwin.apple_sdk.sdk;
+  buildInputs = [ openpam ];
 
   nativeBuildInputs = [ cmake ];
 
```

Reply:
this patch doesn't work for me on sdk 10.12 x64 darwin.
```
/tmp/nix-build-pam_reattach-1.3.drv-0/source/src/xpc.h:29:10: fatal error: 'xpc/xpc.h' file not found
#include <xpc/xpc.h>
```
so i think darwin.apple_sdk.sdk needs to be conditionally included for now.

Reply:
I think this is correct, xpc is currently patched out in the source releases I believe so getting it from the SDK is appropriate.